packages/xo-server/src/xapi-object-to-xo.mjs

```diff
@@ -397,6 +397,14 @@ const TRANSFORMS = {
         }
       })(),
       expNestedHvm: obj.platform['exp-nested-hvm'] === 'true',
+      getFirmwareRecommendations: (() => {
+        const vmRecommendations = obj.recommendations
+        const parsedRecommendations = parseXml(vmRecommendations)
+        const supportsFirmware = parsedRecommendations?.restrictions?.restriction.filter(
+          restriction => restriction.field === 'supports-bios' || restriction.field === 'supports-uefi'
+        )
+        return supportsFirmware
+      })(),
```

Comment:
I think, it would be easier to read and maintain if we only return if the current VM firmware is supported. 
Something like that:

```js
// xapi-object-to-xo
...
isFirmwareSupported: (() => {
  const vmRecommendations = obj.recommendations
  const parsedRecommendations = parseXml(vmRecommendations)
  const unsupportedFirmwares = parsedRecommendations?.restrictions?.restriction
    .filter(
      restriction =>
        (restriction.field === 'supports-bios' || restriction.field === 'supports-uefi') &&
        restriction.value === 'no'
    )
    .map(restriction => restriction.field.split('-')[1])

  return !unsupportedFirmwares?.includes(obj.HVM_boot_params.firmware)
})(),
...
```

And in the `vm/tab-advanced`, you can easly display the warning if necessary like that:
```jsx
// vm/tab-advanded.js
...
{!vm.isFirmwareSupported && (
  <span>
    ...
  </span>
...
```

Reply:
:+1: And since we only want to test one firmware, we can even simplify it more:
```js
isFirmwareSupported: (() => {
  const restrictions = parseXml(obj.recommendations)?.restrictions?.restriction

  if (restrictions === undefined) {
    return true
  }

  const field = `supports-${obj.HVM_boot_params.firmware}`
  const firmwareRestriction = restrictions.find(restriction => restriction.field === field)

  return firmwareRestriction === undefined || firmwareRestriction.value !== 'no'
})()
```